organizations/organizations.go

```diff
@@ -290,3 +290,78 @@ func UpdateLogo(w http.ResponseWriter, r *http.Request) {
 
 	utils.GetSuccess("organization logo updated successfully", nil, w)
 }
+
+
+// transfer workspace ownership
+func TransferOwnership(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	org_collection, member_collection := "organizations", "members"
+
+	org_Id := mux.Vars(r)["id"]
+
+	// Check if organization id is valid
+	orgId, err := primitive.ObjectIDFromHex(org_Id)
+	if err != nil {
+		utils.GetError(errors.New("invalid organization id"), http.StatusBadRequest, w)
+		return
+	}
+
+	// Check if organization id exists in the database
+	orgDoc, _ := utils.GetMongoDbDoc(org_collection, bson.M{"_id": orgId})
+	if orgDoc == nil {
+		utils.GetError(errors.New("organization does not exist"), http.StatusBadRequest, w)
+		return
+	}
+
+	requestData := make(map[string]string)
+	if err := utils.ParseJsonFromRequest(r, &requestData); err != nil {
+		utils.GetError(err, http.StatusUnprocessableEntity, w)
+		return
+	}
+
+	email := requestData["email"]
+	
+	// confirm if email supplied is valid
+	if !utils.IsValidEmail(strings.ToLower(email)) {
+		utils.GetError(errors.New("email is not valid"), http.StatusBadRequest, w)
+		return
+	}
+
+	orgMember, err := FetchMember(bson.M{"email": email})
+
+	if err != nil {
+		utils.GetError(errors.New("user not a member of any work pace"), http.StatusBadRequest, w)
+		return
+	}
+
+	// confirm if the proposed new owner is a member of the organization
+	memberDoc, _ := utils.GetMongoDbDoc(member_collection, bson.M{"org_id": org_Id, "_id": orgMember.ID})
+	if memberDoc == nil {
+		utils.GetError(errors.New("this user is not a member of this work space"), http.StatusBadRequest, w)
+		return
+	}
+
+	if orgMember.Role == "owner" {
+		utils.GetError(errors.New("this user already owns this organization"), http.StatusBadRequest, w)
+		return
+	}
+
+	memID := orgMember.ID
+
+	memberID := memID.Hex()
+
+	updateRes, err := utils.UpdateOneMongoDbDoc(member_collection, memberID, bson.M{"role":"owner"})
```

Comment:
You can easily pass orgMember.ID.Hex() but the value in the filter is supposed to be a primitive not Hex, what happened? You sure this works?

Then please use the constant in the models file for the collection name. Thanks

Reply:
orgMember.ID.Hex()
This looks cleaner

I'll use the constant

Yeah... it works

UpdateOneMongoDbDoc takes in a string for the ID. This was the major source of the error the first time around. OrgMember.ID returns a primitive. Any IDE will straigjt away notify you if you try using primitive for string.

Last time around I used .String() to do the conversion. Further debugging showed it did it for the whole raw objectID hence why i introduced Hex.